shared/middleware/files_api.rb

```diff
@@ -487,10 +487,11 @@ def files_put_file(encrypted_channel_id, filename, body)
     end
     new_entry_hash = JSON.parse new_entry_json
     # Replace downcased filename with original filename (to preserve case in the manifest)
-    new_entry_hash['filename'] = filename
+    new_entry_hash['filename'] = CGI.unescape(filename)
```

Comment:
Thanks for looking in to this. One possible case that is probably not tested is the rename operation where the old name could be unescaped, though in practice that is hard to do since it is a query parameter.

However, the bigger issue that I had is that our test helper method `get_object` doesn't like unescaped paths (probably at the `Rack::Test::Methods` level). So the tests for now verify that creating files with either unescaped or escaped paths will both work equally well - and can be retrieved and deleted using escaped paths.